Initial Search Conversion tracking

robearlam opened the pull request:
This is the first iteration of Conversion Tracking for search. When a user clicks on a search result hosted on the Dev Portal, then scrolls to the bottom of the page, i.e. reads the full article, then a conversion event will be triggered.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update (non-breaking change; modified files are limited to the `/data` directory or other markdown files)

## Checklist:
- [x] I have read the Contributing guide.
- [x] My code/comments/docs fully adhere to the Code of Conduct.
- [x] My change is a code change.
- [ ] My change is a documentation change and there are NO other updates required.
- [ ] My change has new or updated images which are stored in the `/public/images` folder that need to be migrated to Sitecore DAM

** NOTE FOR WHEN TESTING **
The Developer Portal will default to forwarding to the prod domain, so to test the request is made you will need to change the domain back to the Preview Domain

markvanaalst commented:
Looks good to me, I can see the publish event when I scroll down.

Two questions:
1. Should we add the query string parameter when people click a link in the preview search results as well?
2. A lot of pages have "promotional" components like video's at the bottom of each page. Right now it seems that the event is only triggered when reaching the footer. Anyway we can trigger it when the end of the main content has been reached?

robearlam commented:
Good points, I added the tracking param to the PreviewSearch as well.

With the tracking earlier in the doc it's tricky as most of the long form content comes from fairly complex MD files, which makes its complex to have a constant time to trigger the event. I've changed it now to at least exclude the footer which should help a bit?
